service/src/main/java/uk/gov/hmcts/reform/fpl/service/PrepareDirectionsForDataStoreService.java

```diff
@@ -0,0 +1,239 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.enums.ComplyOnBehalfEvent;
+import uk.gov.hmcts.reform.fpl.enums.DirectionAssignee;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Direction;
+import uk.gov.hmcts.reform.fpl.model.DirectionResponse;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.utils.ElementUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang.StringUtils.isEmpty;
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.fpl.enums.ComplyOnBehalfEvent.COMPLY_ON_BEHALF_SDO;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.COURT;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.OTHERS;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.PARENTS_AND_RESPONDENTS;
+
+@Service
+public class PrepareDirectionsForDataStoreService {
```

Comment:
Maybe some high level explanation what types of method can be find here would be useful, at the first glance it would be related to storing in CCD, but then we are firing methods here before generating the doc, so maybe name not 100% correct?

Reply:
hmm, maybe we can split this down more? Have a "mid event" sort of service also? 🤔

Reply:
or maybe it should be renamed to PrepareDirectionsForOrderService?